modules/Findscript.cmake.in

```diff
@@ -88,6 +88,25 @@ find_path(@CMAKE_PROJECT_NAME@_DIR
     "C:/dev/extern/${XP_INSTALLED_NAME}"
   DOC "@CMAKE_PROJECT_NAME@ directory"
   )
+if(NOT @CMAKE_PROJECT_NAME@_DIR)
+  if(EXISTS $ENV{extern_DIR})
+    set(archive_name "${XP_INSTALLED_NAME}.tar.xz")
+    message(STATUS "${XP_INSTALLED_NAME} not found.\nAttempting download of ${archive_name} ...")
+    file(DOWNLOAD @XP_DOWNLOAD_URL@/${@CMAKE_PROJECT_NAME@_REV}/${archive_name}
+      $ENV{extern_DIR}/${archive_name}
+      )
+    message(STATUS "Attempting extraction of ${archive_name} ...")
+    file(ARCHIVE_EXTRACT
+      INPUT $ENV{extern_DIR}/${archive_name}
+      DESTINATION $ENV{extern_DIR}
+      )
+    message(STATUS "Attempting to remove ${archive_name}")
+    file(REMOVE $ENV{extern_DIR}/${archive_name})
+    if(EXISTS $ENV{extern_DIR}/${XP_INSTALLED_NAME})
+      set(@CMAKE_PROJECT_NAME@_DIR $ENV{extern_DIR}/${XP_INSTALLED_NAME})
+    endif()
+  endif()
+endif()
 if(NOT @CMAKE_PROJECT_NAME@_DIR)
   set(@CMAKE_PROJECT_NAME@_INSTALL_INFO @XP_INSTALL_INFO@) # @CMAKE_PROJECT_NAME@ can set(XP_INSTALL_INFO) to define this
```

Comment:
Cameron suggested possibly modifying these error messages down here to say they can set the environment variable to automatically download.

Reply:
I'm not sure if this line would be the best place to do that, or if XP_INSTALL_INFO should be modified somewhere else.